### DISCUSSION TITLE
MatrixSymbols

### DISCUSSION THREAD

**Discussion by RiccardoRossi** [sympy/sympy#20603] [Q&A]
Dear all,

i have a few questions about the use of MatrixSymbol(s)

1. Is there a way to define a symbol as symmetric?
2. Is there an operation to contract two matrix symbols? I mean something that would be normally indicated by the symbol ":" it would be indicated as A:B. 

thx in advance
Riccardo

**Comment by asmeurer** [comment]
Assumptions in the matrices use the new assumptions, which are defined separately from the object (unlike the assumptions on Symbol). So to represent that a matrix symbol `A` is symmetric, use `Q.symmetric(A)`. The function `ask()` and `refine()` can then be used to manipulate this assumption, depending on what you want to do. 

I'm not sure about the second question.

**Reply by VinceECN** [reply]
Interesting. In sympy's doc -> how-to -> assumptions, it says that the new assumptions system is not really used anywhere in sympy, and the "old" one is still preferred. It doesnt give a doc for the new system (and doesnt say that matrices use it), but I assume it is related to this https://docs.sympy.org/latest/modules/assumptions/index.html. However it isnt clear to me how it is meant to be used, and if it is actually accounted for in most sympy operations. From the examples I read it appears to be mostly used a posteriori, to check if a symbol is positive for instance, rather that saying, when defining that symbol, that it will be positive.

**Comment by sylee957** [comment]
I think that the `BlockMatrix` operation can be used to join matrices by rows, columns.
